internal/controller/operator/factory/vlagent/vlagent.go

```diff
@@ -569,6 +707,33 @@ func deleteOrphaned(ctx context.Context, rclient client.Client, cr *vmv1.VLAgent
 			return fmt.Errorf("cannot remove serviceaccount: %w", err)
 		}
 	}
+	if cr.Spec.K8sCollector.Enabled {
+		if err := finalize.SafeDeleteWithFinalizer(ctx, rclient, &appsv1.StatefulSet{ObjectMeta: objMeta}, &owner); err != nil {
+			return fmt.Errorf("cannot remove statefulset: %w", err)
+		}
+	} else {
+		if err := finalize.SafeDeleteWithFinalizer(ctx, rclient, &appsv1.DaemonSet{ObjectMeta: objMeta}, &owner); err != nil {
+			return fmt.Errorf("cannot remove daemonset: %w", err)
+		}
+	}
+	if !cr.IsOwnsServiceAccount() || !cr.Spec.K8sCollector.Enabled {
+		objects := []client.Object{
+			&corev1.ServiceAccount{ObjectMeta: objMeta},
+		}
+		if config.IsClusterWideAccessAllowed() {
+			rbacMeta := metav1.ObjectMeta{Name: cr.GetClusterRoleName(), Namespace: cr.Namespace}
+			objects = append(objects,
+				&rbacv1.ClusterRoleBinding{ObjectMeta: rbacMeta},
+				&rbacv1.ClusterRole{ObjectMeta: rbacMeta},
+			)
+		}
+		owner := cr.AsCRDOwner()
+		for _, o := range objects {
+			if err := finalize.SafeDeleteWithFinalizer(ctx, rclient, o, owner); err != nil {
```

Comment:
WDYT of changing delete for RBAC with `SafeDeleteForSelectorsWithFinalizer`?

 Also, it looks like ServiceAccount already deleted above:

```
	if !cr.IsOwnsServiceAccount() {
		if err := finalize.SafeDeleteWithFinalizer(ctx, rclient, &corev1.ServiceAccount{ObjectMeta: objMeta}, &owner); err != nil {
			return fmt.Errorf("cannot remove serviceaccount: %w", err)
		}
	}
```

Reply:
fixed it